staker-rewards/lib/cache.js

```diff
@@ -0,0 +1,181 @@
+import path from "path"
+import pAll from "p-all"
+import low from "lowdb"
+import FileSync from "lowdb/adapters/FileSync.js"
+
+import { getPastEvents, callWithRetry } from "./contract-helper.js"
+import { KeepStatus, getKeepStatus } from "./keep.js"
+
+const DATA_DIR_PATH = path.resolve(process.env.DATA_DIR_PATH || "./data")
+const CACHE_PATH = path.resolve(DATA_DIR_PATH, "cache.json")
+
+// Our expectation on how deep can chain reorganization be. We need this
+// parameter because the previous cache refresh could store data that
+// are no longer valid due to a chain reorganization. To overcome this
+// problem we lookup `REORG_DEPTH_BLOCKS` earlier than the last refresh
+// block when starting the cache refresh.
+const REORG_DEPTH_BLOCKS = 12
+
+const CONCURRENCY_LIMIT = 3
+
+export default class Cache {
+  constructor(web3, contracts) {
+    this.web3 = web3
+    this.contracts = contracts
+  }
+
+  async initialize() {
+    this.cache = low(new FileSync(CACHE_PATH))
+    await this.cache
+        .defaults({
+          keeps: [],
+          lastRefreshBlock: this.contracts.factoryDeploymentBlock,
+        })
+        .write()
+  }
+
+  async refresh() {
+    const factory = await this.contracts.BondedECDSAKeepFactory.deployed()
+
+    const previousRefreshBlock = this.cache.get("lastRefreshBlock").value()
+
+    const startBlock =
+        previousRefreshBlock - REORG_DEPTH_BLOCKS > 0
+            ? previousRefreshBlock - REORG_DEPTH_BLOCKS
+            : 0
+
+    console.log(
+        `Looking for keeps created since block [${startBlock}]...`
+    )
+
+    const keepCreatedEvents = await getPastEvents(
+        this.web3,
+        factory,
+        "BondedECDSAKeepCreated",
+        startBlock
+    )
+
+    const newKeeps = []
+    keepCreatedEvents.forEach((event) => {
+      newKeeps.push({
+        address: event.returnValues.keepAddress,
+        members: event.returnValues.members,
+        creationBlock: event.blockNumber,
+      })
+    })
+
+    const cachedKeepsCount = this.cache.get("keeps").value().length
+
+    console.log(
+        `Number of keeps created since block ` +
+        `[${startBlock}]: ${newKeeps.length}`
+    )
+
+    console.log(`Number of keeps in the cache: ${cachedKeepsCount}`)
+
+    const actions = []
+    newKeeps.forEach((keep) => {
+      const currentlyCached = this.cache
```

Comment:
`s/currentlyCached/isInCache (isCached)`? 

Keep once cached is cached forever so "currently" does not fit well here IMO.